microceph/cmd/microcephd/main.go

```diff
@@ -63,33 +62,42 @@ func (c *cmdDaemon) Command() *cobra.Command {
 }
 
 func (c *cmdDaemon) Run(cmd *cobra.Command, args []string) error {
-	m, err := microcluster.App(microcluster.Args{StateDir: c.flagStateDir, Verbose: c.global.flagLogVerbose, Debug: c.global.flagLogDebug})
+	m, err := microcluster.App(microcluster.Args{StateDir: c.flagStateDir})
 	if err != nil {
 		return err
 	}
 
-	h := &config.Hooks{}
-	h.PostBootstrap = func(s *state.State, initConfig map[string]string) error {
+	h := &state.Hooks{}
+	h.PostBootstrap = func(ctx context.Context, s state.State, initConfig map[string]string) error {
 		data := common.BootstrapConfig{}
 		interf := interfaces.CephState{State: s}
 		common.DecodeBootstrapConfig(initConfig, &data)
-		return ceph.Bootstrap(interf, data)
+		return ceph.Bootstrap(ctx, interf, data)
 	}
 
-	h.PostJoin = func(s *state.State, initConfig map[string]string) error {
+	h.PostJoin = func(ctx context.Context, s state.State, initConfig map[string]string) error {
 		interf := interfaces.CephState{State: s}
-		return ceph.Join(interf)
+		return ceph.Join(ctx, interf)
 	}
 
-	h.OnStart = func(s *state.State) error {
+	h.OnStart = func(ctx context.Context, s state.State) error {
 		interf := interfaces.CephState{State: s}
-		return ceph.Start(interf)
+		return ceph.Start(ctx, interf)
 	}
 
 	h.PreRemove = ceph.PreRemove(m)
 
-	m.AddServers(api.Servers)
-	return m.Start(context.Background(), database.SchemaExtensions, nil, h)
+	daemonArgs := microcluster.DaemonArgs{
+		Verbose:          c.global.flagLogVerbose,
+		Debug:            c.global.flagLogDebug,
+		Version:          "UNKNOWN", // FIXME: Add an explicit version to MicroCeph.
```

Comment:
@masnax can you please add some comments to describe what versioning is this.

Reply:
When using the MicroCeph API, it is useful to determine the current version of MicroCeph so that compatibility can be verified. We have an endpoint at `/core/1.0` that provides basic identifying information like the cluster name, address, and project version. 

For example, from MicroCloud, we should be able to tell from the API that MicroCeph is installed from a revision belonging to a supported snap track/risk. Like MicroCloud LTS should verify that MicroCeph is also installed as an LTS version. 

Microcluster validates that a version must be present, but I wasn't able to find any value in the project that matched this criteria so I put `UNKOWN` here. 


I'll update the commit with the above information as well.

Reply:
I've added a comment in-line with the `Version` field now.